rbc/heavydb/array.py

```diff
@@ -24,12 +29,39 @@ class HeavyDBArrayType(HeavyDBBufferType):
         bool is_null;
       }
     """
+
+    @property
+    def numba_pointer_type(self):
+        return ArrayPointer
+
     @property
     def buffer_extra_members(self):
         return ('bool is_null',)
 
 
-ArrayPointer = BufferPointer
+class ArrayPointer(BufferPointer):
+    def copy(self, context, builder, val, retptr):
+        from .buffer import memalloc
+        ptr_type = self.dtype.members[0]
+        element_size = int64_t(ptr_type.dtype.bitwidth // 8)
+
+        src = builder.extract_value(builder.load(val), 0)
+        element_count = builder.extract_value(builder.load(val), 1)
+        is_null = builder.extract_value(builder.load(val), 2)
```

Comment:
Sort of unrelated to this PR, but should we start naming our IR instructions? I think it makes the IR a lot more readable during debugging.

For instance, these could be:

```python
src = builder.extract_value(builder.load(val, "array_struct"), 0, "array_buff_ptr")
element_count = builder.extract_value(builder.load(val, "array_struct"), 1, "array_size")
is_null = builder.extract_value(builder.load(val, "array_struct"), 2, "array_is_null")
```

It would also help separate what is rbc-generated IR from numba-generated IR. I think as long as the instruction supports it, we should name it to reflect its counterpart in the code, so that the IR is close to as readable as the code that generates it.

Not only is it a bonus for readability, but I think it helps a lot in creating an "index" for generated IR instructions. E.g., if you're trying to look for which code generated some IR, you can CTRL+F for its name and usually find the corresponding builder call that emitted it.